src/lib/hooks/useCurrencyColor.ts

```diff
@@ -0,0 +1,78 @@
+import { Currency } from '@uniswap/sdk-core'
+import { useTheme } from 'lib/theme'
+import Vibrant from 'node-vibrant/lib/bundle'
+import { useEffect, useLayoutEffect, useState } from 'react'
+
+import useCurrencyLogoURIs from './useCurrencyLogoURIs'
+
+const colors = new Map<string, string | undefined>()
+
+/**
+ * Extracts the prominent color from a token.
+ * NB: If cached, this function returns synchronously; using a callback allows sync or async returns.
+ */
+async function getColorFromLogoURIs(logoURIs: string[], cb: (color: string | undefined) => void = () => void 0) {
+  const key = logoURIs[0]
+  let color = colors.get(key)
+
+  if (!color) {
+    for (const logoURI of logoURIs) {
+      let uri = logoURI
+      if (logoURI.startsWith('http')) {
+        // Color extraction must use a CORS-compatible resource, but the resource may already be cached.
+        // Adds a dummy parameter to force a different browser resource cache entry. Without this, color extraction prevents resource caching.
+        uri += '?color'
+      }
+
+      color = await getColorFromUriPath(uri)
+      if (color) break
+    }
+  }
+
+  colors.set(key, color)
+  return cb(color)
+}
+
+async function getColorFromUriPath(uri: string): Promise<string | undefined> {
+  try {
+    const palette = await Vibrant.from(uri).getPalette()
+    return palette.Vibrant?.hex
+  } catch {}
```

Comment:
should we use this?

```
let v = Vibrant.from(src)
  .useQuantizer(Vibrant.Quantizer.WebWorker)
```

Reply:
I don't think the logos are complex/large enough images to need this - I think it already gets the color within a frame or two, and a webworker would not be able to beat that (because it already requires an additional frame or two to communicate with the webworker).